src/dmd/lexer.d

```diff
@@ -590,8 +665,12 @@ class Lexer
                     }
                     if (commentToken)
                     {
-                        p++;
-                        endOfLine();
+                        version (DMDLIB) {}
+                        else
+                        {
+                            p++;
+                            endOfLine();
+                        }
```

Comment:
This should only be skipped when `whitespaceToken` is `true`?

Reply:
Yes, but actually no. Suppose we have
```
// some comment(implicit '\n')
some statement
```
and the current iteration is k.

Without `version DMDLIB`:
k / TOK.comment and p is incremented past the '\n'
k+1 / TOK.statement

With `version DMDLIB`, _as it is now_:
- if `whitespaceToken` is `true`:
k / TOK.comment and p is pointing to the '\n'
k+1 / TOK.whitespace
k+2 / TOK.statement
- if `whitespaceToken` is `false`:
k / TOK.comment and p is pointing to the '\n'
k+1 / hits `case '\n'`, (but we don't lex whitespaces) increments p and continues
k+2 / TOK.statement

If we had something like
```
version (DMDLIB)
{
    if (whitespaceToken == false)
    {
        p++;
        endOfLine();
    }
}
else
{
    p++;
    endOfLine();
}
```
- if `whitespaceToken` is `true`:
same as above
- if `whitespaceToken` is `false`:
k / TOK.comment and p is incremented past the '\n'
k+1 / TOK.statement

Basically, the difference boils down to what we would prefer: one more if-statement to check or one more case to match.